### DISCUSSION TITLE
MongoDB Panache ObjectId in native image mode

### DISCUSSION THREAD

**Discussion by szabolcs-dekany** [quarkusio/quarkus#32511] [Q&A]
Hey all! I'm currently having in issue with running my simple little app in native image mode.

I have the following simple entity:
```java
@Builder // Lombok
@MongoEntity(collection = "entity")
public class SomeEntity extends PanacheMongoEntity {
  public LocalDate createdAt;
}
```

I have to simple transactional method that creates an entity of this type:
```java
@Transactional
  public String createEntity() {

    var entity = SomeEntity.builder()
        .createdAt(LocalDateTime.now(ZoneId.of("UTC"));)
        .build();
    entity.persist();
    return entity.id.toString();
  }
  ```

This setup works fine in regular JVM mode, an ObjectId is created automatically for the entity, but as soon as I compile it into a native image, this seem to not work and I'm getting NPs when trying to call `return entity.id.toString();` 

Tried to see if its Lombok, tried it without the builder but it seemingly makes no difference. Anyone ran into this before?

Cheers,
Szabolcs

[a comment by an automated account is not shown]

**Comment by loicmathieu** [comment]
This is strange, it seems your Entity is not included in the native image but we register all of them.
Can you create a reproducer and open an issue ?
